File: Interview Practice/Arrays/firstNotRepeatingCharacter/first_not_repeating_character.c
char firstNotRepeatingCharacter(char * s) {
	int appeared[26] = {0};

	for (int i = 0; i < strlen(s); i++) {
		appeared[s[i]-'a']++;
	}

	for (int i = 0; i < strlen(s); i++) {
		if (appeared[s[i]-'a'] == 1) {
			return s[i];
		}
	}

	return '_';
}

File: Interview Practice/Arrays/firstNotRepeatingCharacter/first_not_repeating_character.py
def firstNotRepeatingCharacter(s):
	appeared = [0]*26

	for c in s:
		appeared[ord(c) - ord('a')] += 1

	for i in range(len(s)):
		if (appeared[ord(s[i]) - ord('a')] == 1):
			return s[i]

	return '_'
